### ISSUE TITLE
Mac OS extlibs have hard-coded absolute paths

### ISSUE THREAD

**Issue by jqdg** [SFML/SFML#1815]
Some of the bundled extlibs for Mac OS have hard-coded absolute paths (rather than relative to `@rpath`). For example,

```
$ otool -L extlibs/libs-osx/Frameworks/FLAC.framework/FLAC                                                                                                         
extlibs/libs-osx/Frameworks/FLAC.framework/FLAC:
	@rpath/../Frameworks/FLAC.framework/Versions/A/FLAC (compatibility version 12.0.0, current version 12.0.0)
	/Users/sfml/Desktop/deps/build_deps/libogg-1.3.3/build/1.3.3/x86_64/root/lib/libogg.0.dylib (compatibility version 9.0.0, current version 9.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.0.0)
```

Note the path `/Usrs/sfml/Desktop/...`. That path should instead be something like `@rpath/../Frameworks/ogg.framework/Versions/A/ogg`.

This will result in a runtime error

```
dyld: Library not loaded: /Users/sfml/Desktop/deps/build_deps/libogg-1.3.3/build/1.3.3/x86_64/root/lib/libogg.0.dylib
```

unless the user happens to have ogg and vorbis installed in their runtime library search path (in `/usr/local/lib` or similar, for example through homebrew).

This affects the FLAC, vorbis, vorbisenc, and vorbisfile libraries.

**Comment by andrew-r-king** [comment]
I noticed this recently too (hit this while signing an sfml app) and have some thoughts:

The libogg and libvorbis dylibs should definitely be included, as they also need to be fat/universal (x86_64/arm64), and Homebrew only gets dylibs for the architecture of your system right now, so if they weren't included, you'd have to build them anyway.

**Comment by eXpl0it3r** [comment]
Looking at it closer it doesn't seem to actually be a regression, as the same path also exists for 2.5.1 extlibs.
I'll see what arguments need to be set to get it fixed.

**Comment by eXpl0it3r** [comment]
Can you give #1843 a try?

**Comment by andrew-r-king** [comment]
Seems to be fixed for me! I signed both x86_64 and universal app bundles and they launched correctly.

**Comment by eXpl0it3r** [comment]
Fixed with 600d0f3baf3c1592d2f2c10e4d53daa940bcb0f4
